Source/WebCore/dom/Element.cpp

```diff
@@ -5363,4 +5374,40 @@ bool Element::isPopoverShowing() const
     return popoverData() && popoverData()->visibilityState() == PopoverVisibilityState::Showing;
 }
 
+// https://drafts.csswg.org/css-contain/#relevant-to-the-user
+bool Element::isRelevantToUser() const
+{
+    return hasRareData() && elementRareData()->contentRelevancyStatus();
+}
+
+OptionSet<ContentRelevancyStatus> Element::contentRelevancyStatus() const
+{
+    if (!hasRareData())
+        return { };
+    return elementRareData()->contentRelevancyStatus();
+}
+
+void Element::updateContentRelevancyStatus(ContentRelevancyStatus status, bool value)
+{
+    if (contentRelevancyStatus().contains(status) != value) {
+        ensureElementRareData().setContentRelevancyStatus(status, value);
+        invalidateStyle();
+    }
+}
```

Comment:
It's a bit confusing that `updateContentRelevancyStatusIfNeeded` walks over the parent nodes when this method doesn't.

Also, can we set this up in a way that makes it clear when `updateContentRelevancyStatusIfNeeded` should be called, and when `updateContentRelevancyStatus` should be called?

E.g. if `updateContentRelevancyStatusIfNeeded` should always be called, it might be preferable to make this a static function.

Reply:
I can see that the names can be confusing, and the "IfNeeded" part is more of an implementation detail, so I renamed these two methods to hopefully make things more clear.